packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -62,6 +63,123 @@ const TextStyle _kToolbarButtonDisabledFontStyle = TextStyle(
 // Eyeballed value.
 const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 
+// Generates the child that's passed into CupertinoTextSelectionToolbar.
+class _CupertinoTextSelectionToolbarWrapper extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarWrapper({
+    Key key,
+    this.arrowTipX,
+    this.barTopY,
+    this.clipboardStatus,
+    this.handleCut,
+    this.handleCopy,
+    this.handlePaste,
+    this.handleSelectAll,
+    this.isArrowPointingDown,
+  }) : super(key: key);
+
+  final double arrowTipX;
+  final double barTopY;
+  final ClipboardStatus clipboardStatus;
+  final VoidCallback handleCut;
+  final VoidCallback handleCopy;
+  final VoidCallback handlePaste;
+  final VoidCallback handleSelectAll;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarWrapperState createState() => _CupertinoTextSelectionToolbarWrapperState();
+}
+
+class _CupertinoTextSelectionToolbarWrapperState extends State<_CupertinoTextSelectionToolbarWrapper> {
+  // Inform the widget that the value of clipboardStatus has changed.
+  void _onChangedClipboardStatus() {
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    widget.clipboardStatus.addListener(_onChangedClipboardStatus);
+    widget.clipboardStatus.update();
+  }
+
+  @override
+  void didUpdateWidget(_CupertinoTextSelectionToolbarWrapper oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.handlePaste != null) {
```

Comment:
Do we need to update every time? Or only when we switch from null to non-null?

Reply:
I've been erring on the side of updating too frequently in order to help handle cases where the clipboard can change that I'm not thinking of.

The typical place where didUpdateWidget is called is when the visible text selection menu changes, like when pressing "Select All", or opening the overflow menu.  In most normal uses the user probably isn't going to be able to change the content of their clipboard between opening the selection menu and pressing "Select All" (without backgrounding the app), in which case your optimization would be fine.  However, I worry that users could use a physical keyboard and press ctrl-c, or use Gboard and copy something out of the search field inside of Gboard, or something else I'm not thinking of.

I'm happy to add the optimization if you're not worried, though.  Really, I wish that `Clipboard` had a way to directly listen to changes!